monitoring/list_resources.js

```diff
@@ -0,0 +1,165 @@
+    var monitoring = google.monitoring('v3');
+    monitoring.projects.monitoredResourceDescriptors.list({
+        auth: authClient,
+        name: projectId,
+    }, function (error, monitoredResources) {
+        if (error) {
+            console.log('Error Retrieving Monitored Resource Descriptors', error);
+            return;
+        }
+        console.log('listMonitoredResourceDescriptors: ');
+        console.log(monitoredResources);
+        callback();
+    });
+}
+
+/**
+ * This Lists the metric descriptors that start with our METRIC name, in this case the
+ * CPU usage time.
+ * @param {googleAuthClient} authClient - The authenticated Google api client
+ * @param {String} projectId - the project id
+ * @param {requestCallback} callback - a function to be called when the server
+ *     responds with the list of monitored resource descriptors
+ */
+function listMetricDescriptors(authClient, projectId, callback) {
+    var monitoring = google.monitoring('v3');
+    monitoring.projects.metricDescriptors.list({
+        auth: authClient,
+        filter: 'metric.name="' + METRIC + '"',
+        name: projectId
+    }, function (error, metricDescriptors) {
+        if (error) {
+            console.log('Error Retrieving Metric Descriptors', error);
+            return;
+        }
+        console.log('listMetricDescriptors');
+        console.log(metricDescriptors);
+        callback();
+    });
+}
+
+/**
+ * This Lists all the timeseries created between START_TIME and END_TIME
+ * for our METRIC.
+ * @param {googleAuthClient} authClient - The authenticated Google api client
+ * @param {String} projectId - the project id
+ * @param {requestCallback} callback - a function to be called when the server
+ *     responds with the list of monitored resource descriptors
+ */
+function listTimeseries(authClient, projectId, callback) {
+    var monitoring = google.monitoring('v3');
+    var startTime = getStartTime();
+    var endTime = getEndTime();
+
+    monitoring.projects.timeSeries.list({
+        auth: authClient,
+        filter: 'metric.type="' + METRIC + '"',
+        pageSize: 3,
+        'interval.startTime': startTime,
+        'interval.endTime': endTime,
+        name: projectId
+    }, function (error, timeSeries) {
+        if (error) {
+            console.log('Error Retrieving Timeseries', error);
+            return;
+        }
+        console.log('listTimeseries');
+        console.log(timeSeries);
+        callback();
+    });
+}
+
+
+google.auth.getApplicationDefault(function (error, authClient) {
+    if (error) {
+        console.log(error);
+        process.exit();
+    }
+
+    // Depending on the environment that provides the default credentials
+    // (e.g. Compute Engine, App Engine), the credentials retrieved may require
+    // you to specify the scopes you need explicitly.
+    // Check for this case, and inject the Cloud Storage scope if required.
+    if (authClient.createScopedRequired &&
```

Comment:
I don't think it's done for you:

https://github.com/google/google-auth-library-nodejs#example-usage